config/v1/types_console.go

```diff
@@ -60,3 +62,19 @@ type ConsoleAuthentication struct {
 	// +kubebuilder:validation:Pattern=`^$|^((https):\/\/?)[^\s()<>]+(?:\([\w\d]+\)|([^[:punct:]\s]|\/?))$`
 	LogoutRedirect string `json:"logoutRedirect,omitempty"`
 }
+
+// Helm configuration
+type HelmConfig struct {
+
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	ChartRepositories []HelmChartRepository `json:"chartRepositories"`
```

Comment:
Do you want `,omitempty`?  I assume chart repositories will always be necessary, even if there were other fields.

Reply:
Wondering if we should mark it as optional, to avoid breaking any existing scripts we might have which doesn't assume this field to be present?

```
// ...
// +optional
ChartRepositories *[]HelmChartRepository `json:"chartRepositories",omitempty`
```

On the console operator side,
```
if spec.ChartRepositories == nil {
   // implies that the field was skipped in the CR
   // the code could default the chart repo to RHD chart repo
}
else if  spec.ChartRepositories != nil && len(spec.ChartRepositories) == 0 {
   // implies that helm charts have to be entirely disabled?
}
```
Or do you think we should keep things simple and populate this field in the console-operator's manifests? Explicit better then implicit :-)

Reply:
Yes, we can make it as robust as possible, but would be nice if console-operator can add missing pieces.